### PR TITLE
feature: create a button for dark/light mode

### LINKED ISSUE #65: [Task] theme toggle switch
it'd be nice to have a theme change toggle switch in the navbar like this:

download

### PR DESCRIPTION
**Pull request by HusseinSerag**
#### Description

Please include a brief summary of the change and include the relevant issue(s).

Closes #65 

#### What change does this PR introduce?

Please select the relevant option(s).

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] CI/CD (updates related to the CI/CD process)
- [ ] Documentation update (changes to docs/code comments)
- [ ] Chore (miscellaneous tasks that do not fall into the above options)

#### What is the proposed approach?

Please give a short overview/explanation on the approach taken to resolve the issue(s).

#### Checklist:

- [ ] The commit message follows our adopted [guidelines](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Testing has been done for the change(s) added (for bug fixes/features)
- [ ] Relevant comments/docs have been added/updated (for bug fixes/features)

### CODE REVIEW

**Comment by HusseinSerag** [comment]
css files could be colocated with the toggle button but that would break our folder structure thus why it is in index.css

**Comment by HusseinSerag** [comment]
which component package can do this?

**Comment by hunxjunedo** [comment]
though it doesn't look exactly like the one in the issue, it actually looks cool:  https://www.npmjs.com/package/react-toggle-dark-mode   . This one also looks good, it looks exactly like the one in the issue: https://github.com/todd-elvers/react-dark-mode-toggle-2

**Comment by HusseinSerag** [comment]
the new button looks good

File: src/components/NavigationBar/AppThemeToggle.tsx [hunk 1]
```diff
@@ -12,12 +12,17 @@ const AppThemeToggle = () => {
 	const handleClick = () => {
 		toggleAppTheme()
 	}
+  
+	const isDark = appTheme === 'dark';
+	return <DarkModeSwitch
+		checked={isDark}
+		onChange={handleClick}
+		size={30}
```
Comment (hunxjunedo) [inline src/components/NavigationBar/AppThemeToggle.tsx:20]: `size={20}`

File: src/components/NavigationBar/AppThemeToggle.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { useAppTheme } from "../../context/AppThemeContext";
```
Comment (hunxjunedo) [inline src/components/NavigationBar/AppThemeToggle.tsx:1]: add this in `navigationBar.tsx`:  `<li className='grid items-center'>
					<AppThemeToggle />
				</li>`   to center the icon

File: tailwind.config.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /** @type {import('tailwindcss').Config} */
 export default {
   content: ["./src/**/*.{js,jsx,ts,tsx}"],
-  darkMode: "dark",
+  darkMode: "class",
```
Comment (hunxjunedo) [inline tailwind.config.js:4]: is this needed ?

Comment (HusseinSerag) [reply tailwind.config.js:4]: Dark mode wasn’t working without this addition. I got it from the docs of course.

Comment (HusseinSerag) [reply tailwind.config.js:4]: I will check if it works without it, and try to understand in general what this property do

Comment (HusseinSerag) [reply tailwind.config.js:4]: Hey, i looked up the docs a couple times, I looked everywhere i didnt find a darkMode: "dark" . There is several darkMode variants but dark isnt one of them. I will resolve, if u find anything else please open it up again

Comment (hunxjunedo) [reply tailwind.config.js:4]: Sure, if you face problems just leave it, we're planning a separate PR for all the theming issues.

**Comment by HusseinSerag** [comment]
Regarding theme of home screen. It is mixed up. As you said another pr requests for all theming problem would do, but the button is done.

**Review by hunxjunedo — APPROVED** [review]

**Pull request merged**